Examples/VaporExample/README.md

```diff
@@ -1,13 +1,107 @@
+It is built using a small library we've written called [mongodb-vapor](https://github.com/mongodb/mongodb-vapor) which includes some helpful code for integrating the two.
 
-To test it out, do the following:
-1. Run `mongod` to start MongoDB running on `localhost:27017`.
-1. Navigate to the `Examples/` directory (one level up from this one.)
-1. Run `../loadExampleData.sh` to load sample data into the database.
-1. Navigate to the root directory of this example.
-1. Run `swift run`.
-1. Navigate to `localhost:8080/kittens` to see the example data loaded on the web page.
+This application is intended to demonstrate best practices for integrating the driver into your backend. It is **not** production-ready, and does not necessarily follow best HTML or Javascript practices. 
+
+The application contains both a REST API as well as a minimal frontend built with Vapor's templating language [Leaf](https://github.com/vapor/leaf).
+
+This application require Swift 5.2 and MongoDB 3.6+. It will run on Linux as well as macOS 10.15+.
+
+## Building and Running the Application
+1. Install MongoDB on your system if you haven't already. Downloads are available [here](https://www.mongodb.com/download-center/community).
+1. Start up MongoDB running locally: `mongod --dbpath some-directory-here`. You may need to specify a `dbpath` directory for the database to use.
+1. Run `./loadData.sh` to load example application data into the database.
+1. Install Swift 5.2+ on your system if you haven't already. You can download Swift and find instructions for installing it [here](https://swift.org/download/).
+1. From the root directory of the project, run `swift build`. This will likely take a while the first time you do so.
+1. Once building has completed, run `swift run` from the root directory. You should get a message that the server has started running on `http://127.0.0.1:8080`.
+1. Open up your browser and visit `http://127.0.0.1:8080`. You should see the application and be able to test out adding, deleting, and editing data in the collection.
+
+## Application Architecture
+
+This is a fully asynchronous application. At its core is [SwiftNIO](https://github.com/apple/swift-nio), which is used to implement both Vapor and the MongoDB driver.
+
+The application has both web and REST API interfaces, which support storing a list of kittens and details about them.
+
+The server will handle the following types of web requests:
+1. A GET request at the root URL `/` loads the main index page containing a list of kittens.
+1. A GET request at the URL `/kittens/{name}` loads a web page with information about the kitten with the specified name.
+
+And the following types of API requests:
+1. A GET request at the URL `/rest` returns a list of kittens.
+1. A POST request at the URL `/rest` adds a new kitten.
+1. A GET request at the URL `rest/kittens/{name}` returns information about the kitten with the specified name.
```

Comment:
a few of these are missing the leading `/`

Reply:
fixed